src/main/java/org/cloudfoundry/promregator/auth/CertificateBasedTokenFetcher.java

```diff
@@ -0,0 +1,55 @@
+package org.cloudfoundry.promregator.auth;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.cloudfoundry.promregator.auth.OAuth2XSUAAEnricher.TokenResponse;
+import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.security.mtls.SSLContextFactory;
+
+public class CertificateBasedTokenFetcher extends TokenFetcher {
+
+	private static final Logger log = LoggerFactory.getLogger(CertificateBasedTokenFetcher.class);
+
+	public CertificateBasedTokenFetcher(OAuth2XSUAAAuthenticationConfiguration authConfig)
+			throws GeneralSecurityException, IOException {
+		super(authConfig);
+	}
+
+	@Override
+	protected final CloseableHttpClient prepareHttpCient(OAuth2XSUAAAuthenticationConfiguration authConfig)
+			throws GeneralSecurityException, IOException {
+		SSLContext sslContext = SSLContextFactory.getInstance().create(this.authConfig.getClient_certificates(),
+				this.authConfig.getClient_key());
+		return HttpClients.custom().setSSLContext(sslContext).build();
+	}
+
+	public TokenResponse getJWT(RequestConfig config) throws IOException {
+
+		if (this.authConfig.getClient_id().contains(":")) {
```

Comment:
A little unclear to me why a colon in the user name should be an (injection?) problem here: In case of certificate-based authentication no HTTP Authentication header is in use.

Reply:
Obsolet. The class does not exist anymore.